ccl/streamingccl/streamingest: TestTenantStreamingShowTenant failed

ccl/streamingccl/streamingest.TestTenantStreamingShowTenant [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/8300211?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/8300211?buildTab=artifacts#/) on master @ [a3d53668ff52c2306f37fcad8e9212ff665ca115](https://github.com/cockroachdb/cockroach/commits/a3d53668ff52c2306f37fcad8e9212ff665ca115):


```
=== RUN   TestTenantStreamingShowTenant
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/90c1b75d835f45b8488807abb5b1092d/logTestTenantStreamingShowTenant2276514814
    test_log_scope.go:79: use -show-logs to present logs inline
    replication_stream_e2e_test.go:958: 
        	Error Trace:	/home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3624/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/streamingccl/streamingest/streamingest_test_/streamingest_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/streamingccl/streamingest/replication_stream_e2e_test.go:958
        	Error:      	Not equal: 
        	            	expected: hlc.Timestamp{WallTime:1673682872240016781, Logical:1, Synthetic:false}
        	            	actual  : hlc.Timestamp{WallTime:1673682872240016781, Logical:0, Synthetic:false}
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -2,3 +2,3 @@
        	            	  WallTime: (int64) 1673682872240016781,
        	            	- Logical: (int32) 1,
        	            	+ Logical: (int32) 0,
        	            	  Synthetic: (bool) false
        	Test:       	TestTenantStreamingShowTenant
    panic.go:522: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/90c1b75d835f45b8488807abb5b1092d/logTestTenantStreamingShowTenant2276514814
--- FAIL: TestTenantStreamingShowTenant (9.72s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/tenant-streaming
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTenantStreamingShowTenant.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-23393

ccl/streamingccl/streamingest.TestTenantStreamingShowTenant [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/8335884?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/8335884?buildTab=artifacts#/) on master @ [43cd78ecd87591d9b1d3b142f29ddfd76013b276](https://github.com/cockroachdb/cockroach/commits/43cd78ecd87591d9b1d3b142f29ddfd76013b276):


```
=== RUN   TestTenantStreamingShowTenant
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/90c1b75d835f45b8488807abb5b1092d/logTestTenantStreamingShowTenant1475603344
    test_log_scope.go:79: use -show-logs to present logs inline
    replication_stream_e2e_test.go:958: 
        	Error Trace:	/home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3222/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/streamingccl/streamingest/streamingest_test_/streamingest_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/streamingccl/streamingest/replication_stream_e2e_test.go:958
        	Error:      	Not equal: 
        	            	expected: hlc.Timestamp{WallTime:1674024009539026281, Logical:1, Synthetic:false}
        	            	actual  : hlc.Timestamp{WallTime:1674024009539026281, Logical:0, Synthetic:false}
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -2,3 +2,3 @@
        	            	  WallTime: (int64) 1674024009539026281,
        	            	- Logical: (int32) 1,
        	            	+ Logical: (int32) 0,
        	            	  Synthetic: (bool) false
        	Test:       	TestTenantStreamingShowTenant
    panic.go:522: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/90c1b75d835f45b8488807abb5b1092d/logTestTenantStreamingShowTenant1475603344
--- FAIL: TestTenantStreamingShowTenant (10.80s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTenantStreamingShowTenant.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/ccl/streamingccl/streamingest/replication_stream_e2e_test.go
+++ b/pkg/ccl/streamingccl/streamingest/replication_stream_e2e_test.go
@@ -945,11 +945,12 @@ func TestTenantStreamingShowTenant(t *testing.T) {
 	require.GreaterOrEqual(t, protectedTime, replicationDetails.ReplicationStartTime.GoTime())
 	require.Nil(t, cutoverTime)
 
-	// Verify the SHOW command prints the right cutover timestamp.
-	futureTime := c.DestSysServer.Clock().Now().Add(24*time.Hour.Nanoseconds(), 0)
+	// Verify the SHOW command prints the right cutover timestamp. Adding some
+	// logical component to make sure we handle it correctly.
+	futureTime := c.DestSysServer.Clock().Now().Add(24*time.Hour.Nanoseconds(), 7)
 	var cutoverStr string
 	c.DestSysSQL.QueryRow(c.T, `ALTER TENANT $1 COMPLETE REPLICATION TO SYSTEM TIME $2::string`,
-		c.Args.DestTenantName, futureTime.GoTime()).Scan(&cutoverStr)
+		c.Args.DestTenantName, futureTime.AsOfSystemTime()).Scan(&cutoverStr)
 	var showCutover string
 	c.DestSysSQL.QueryRow(c.T, fmt.Sprintf("SELECT cutover_time FROM [SHOW TENANT %s WITH REPLICATION STATUS]",
 		c.Args.DestTenantName)).Scan(&showCutover)

